FreeRTOS-Plus/Demo/Common/Logging/windows/Logging_WinSim.c

```diff
@@ -149,6 +149,39 @@ static size_t ulSizeOfLoggingFile = 0ul;
 Socket_t xPrintSocket = FREERTOS_INVALID_SOCKET;
 struct freertos_sockaddr xPrintUDPAddress;
 
+/* The logging thread handle. */
+HANDLE pvLoggingThread = NULL;
+
+/* Windows event used to stop the logging thread and flush the logging buffer. */
+static void * pvLoggingThreadExitEvent = NULL;
+
+/*-----------------------------------------------------------*/
+
+static BaseType_t prvStrEndedWithLineBreak( const char * pcStr )
+{
+    BaseType_t xReturn;
+    size_t uxStrLen = strlen( pcStr );
```

Comment:
This private function is only called in `vLoggingPrintf`. The same problem happens in vLoggingPrintf. A better way to prevent this kind of problem is to specify the string length, like snprintf..